acme/acme/messages.py

```diff
@@ -18,31 +49,24 @@ class Error(jose.JSONObjectWithFields, errors.Error):
     :ivar unicode detail:
 
     """
-    ERROR_TYPE_DESCRIPTIONS = dict(
-        ('urn:acme:error:' + name, description) for name, description in (
-            ('badCSR', 'The CSR is unacceptable (e.g., due to a short key)'),
-            ('badNonce', 'The client sent an unacceptable anti-replay nonce'),
-            ('connection', 'The server could not connect to the client to '
-             'verify the domain'),
-            ('dnssec', 'The server could not validate a DNSSEC signed domain'),
-            ('invalidEmail',
-             'The provided email for a registration was invalid'),
-            ('invalidContact',
-             'The provided contact URI was invalid'),
-            ('malformed', 'The request message was malformed'),
-            ('rateLimited', 'There were too many requests of a given type'),
-            ('serverInternal', 'The server experienced an internal error'),
-            ('tls', 'The server experienced a TLS error during domain '
-             'verification'),
-            ('unauthorized', 'The client lacks sufficient authorization'),
-            ('unknownHost', 'The server could not resolve a domain name'),
-        )
-    )
-
     typ = jose.Field('type', omitempty=True, default='about:blank')
     title = jose.Field('title', omitempty=True)
     detail = jose.Field('detail', omitempty=True)
 
+    @classmethod
+    def with_code(cls, code, **kwargs):
+        """Create an Error instance with an ACME Error code.
+
+        :unicode code: An ACME error code, like 'dnssec'.
+        :kwargs: kwargs to pass to Error.
+
+        """
+        if code not in ERROR_CODES:
+            raise ValueError("The supplied code: %s is not an ACME error"
```

Comment:
Perhaps "not a known ACME error code", since sometimes people will be running an old Certbot in a newer universe.